sampler.go

```diff
@@ -174,6 +181,177 @@ func (s *rateLimitingSampler) Equal(other Sampler) bool {
 
 // -----------------------
 
+type lowerBoundSampler struct {
+	rateLimitingSampler Sampler
+	samplingRate        float64
+	tags                []Tag
+}
+
+// NewLowerBoundSampler creates a sampler that samples at most maxTracesPerSecond. It is a wrapper around
+// a rateLimitingSampler with different tags since it's intended to be used by the adaptiveSampler.
+// The sampler.param tag is sampling rate rather than maxTracesPerSecond. This is used to tell the collector
+// that the trace sampled by this sampler used the samplingRate to determine the throughput for the operation.
+func NewLowerBoundSampler(maxTracesPerSecond, samplingRate float64) (Sampler, error) {
```

Comment:
I find this type rather confusing. The self-contained piece of functionality is the combo sampler (min throughput probabilistic sampler) that applies probabilistic sampler first, and if that fails also checks the rate limiting sampler. Separating that piece of functionality apart does not make sense. Note that with such combo sampler you do not need to have the arbitrary "sampling rate" hanging around.